.github/workflows/haskell-ci.yml

```diff
@@ -17,8 +17,23 @@ on:
   - push
   - pull_request
 jobs:
+  tools_changed:
+    continue-on-error: true
+    runs-on: ubuntu-20.04
+    outputs:
+      should_skip: ${{ steps.skip_check.outputs.should_skip }}
+    steps:
+      - id: skip_check
+        uses: fkirc/skip-duplicate-actions@master
+        with:
+          concurrent_skipping: "never"
+          skip_after_successful_duplicate: "true"
+          paths: '["code/**"]'
+          do_not_skip: '["push", "workflow_dispatch", "schedule"]'
   linux:
     name: Haskell-CI - Linux - ${{ matrix.compiler }}
+    needs: tools_changed
+    if: ${{ needs.tools_changed.outputs.should_skip != 'true' }}
```

Comment:
Note that the expected artifact name for this workflow is:
```yaml
with:
      artifact-name: hsec-tools-${{ github.sha }}
```

If we skip the `linux` job when `code/**` did not change, it would fail to upload the artifact, and the `check-advisories` workflow would fail.  To avoid this, we can base the artifact name on the Git tree hash for the `code/` directory, instead of the commit hash.

I'll update this commit with the Git command to query the tree hash (I wrote it down somewhere, just need to find it).

Reply:
IIUC, it's only used later-on to check advisories in this workflow, I'm not sure it's used elsewhere